### PR TITLE
Micropilot integration + private browsing respect

### PR DESCRIPTION
**Pull request by daviddahl**
I still need to clean up some debugging here, so this is not quite a pull request. Also have private browsing cookies being ignored ( issue 12 )

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -212,3 +237,57 @@ function onQuitApplication(event) {
...
+  // let uploadDuration = (86400 * 1000);
+  let uploadDuration = (60 * 300 * 1000);
+  if (!ss.lastupload) {
+    ss.lastupload = Date.now();
+  }
+
+  // Write cookiemonster collected data to the MP study.
+  // We do not use the MicroPilot data storage mechanism as we are summarizing
+  // the data on the fly - the MicroPilot data collection is not ideal for CM
+  Fuse({ start: ss.lastupload,
```
Comment (monicachew) [inline lib/main.js]: let fuse = Fuse to avoid global scope

**Comment by monicachew** [comment]
Awesome! Let me know when you want a review. 

----- Original Message -----

> I still need to clean up some debugging here, so this is not quite a
> pull request. Also have private browsing cookies being ignored (
> issue 12 )
> You can merge this Pull Request by running
> git pull https://github.com/mozilla/cookiemonster
> micropilot-integration
> Or view, comment on, or merge it at:
> https://github.com/mozilla/cookiemonster/pull/22 Commit Summary
> - First attempt at integrating MicroPilot into CookieMonster
> - fixed timestamp for upload, changed Fuse to use setInterval
> - fixed issue 12 (private browsing) and have gotten the Fuse data
>   uploader to work, needs debugging cleanup
>   File Changes
> - M lib/main.js (79)
> - M package.json (24)
> - A packages/README.md (6)
>   Patch Links:
> - https://github.com/mozilla/cookiemonster/pull/22.patch
> - https://github.com/mozilla/cookiemonster/pull/22.diff

File: lib/main.js [hunk 2]
```diff
@@ -206,9 +263,56 @@ exports.onExamineResponse = onExamineResponse;
...
+           monitor.record(cookiemonster).then(function _record(aData) {
+             monitor.upload(kUPLOAD_URL).then(function _clear() {
+               monitor.clear();
+               ss.lastupload = Date.now();
+             });
+           });
+         },
+         resolve_this: self }).then(function _afterFuse() {
+           // re-initialize cookiemonster data
+           // XXXddahl: this does not reset the stored values!
```
Comment (monicachew) [inline lib/main.js:299]: No -- another reason to switch to monitor.record() for everything, so monitor.clear() works.

**Comment by monicachew** [comment]
Looking good! I'm happy to merge this in, if you're comfortable with it.

**Comment by daviddahl** [comment]
ok!

**Pull request merged**